test/e2e/plugin_lifecycle/plugin_group_lifecycle_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// plugin provides plugin command specific E2E test cases
+package plugin
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-cli/test/e2e/framework"
+)
+
+// This test suite covers plugin group life cycle use cases for central repository
+// it uses local central repo to discovery plugins and plugin groups, for which we need to make sure that
+// docker is running and also local central repo is running, start with 'make start-test-central-repo'
+// we need to update PATH with tanzu binary
+// run the tests with make target 'make start-test-central-repo',
+// this make target by default updates the local central repository URL to environment variable TANZU_CLI_E2E_TEST_LOCAL_CENTRAL_REPO_URL
+// the use case being covered in this suite are:
+//  1. 'plugin group search' there should be some groups always
+//  2. 'tanzu plugin install <plugin|all> --group <group>'
+//     'tanzu plugin install all --group <group>' will be validated by getting list of plugins belongs to specific plugin group by mapping
+//     plugin group version ( vmware-tmc/v9.9.9 ) with plugin version from the plugin search output, tanzu plugin group get <group> is
+//     not supported yet.
+//
+// Use cases not covered:
+// 1. 'tanzu plugin group get <group>'
+// 2. 'tanzu plugin group describe <group>'
+
+var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Plugin-Group-lifecycle]", func() {
+	// Use cases:
+	// a. clean, install one plugin from a plugin group and validate the installation by running plugin describe.
+	// b. install all plugins in a group (to make sure we should be able to install all plugins in a group even when some plugins in group already installed) and validate the installation by running plugin describe for all plugins in a plugin group.
+	Context("plugin install from group: install a plugin from a specific plugin group", func() {
+		// Test case: clean plugins if any installed already
+		It("clean plugins if any installed already", func() {
+			err := tf.PluginCmd.CleanPlugins()
+			Expect(err).To(BeNil(), "should not get any error for plugin clean")
+			pluginsList, err := tf.PluginCmd.ListPlugins()
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(len(pluginsList)).Should(BeNumerically("==", 0), "plugins list should not return any plugins after plugin clean")
+		})
+		// Test case: install a plugin from each plugin group and validate the installation
+		It("install a plugin from each plugin group and validate the plugin installation by 'plugin describe'", func() {
+			for pg := range pluginGroupToPluginListMap {
+				plugins := pluginGroupToPluginListMap[pg]
+				err := tf.PluginCmd.InstallPluginsFromGroup(plugins[0].Name, pg)
+				Expect(err).To(BeNil(), "should not get any error for plugin install from plugin group")
```

Comment:
Highly unlikely but what if a plugin group contains the same named plugin with different targets? 

For example, a plugin group `vmware-tkg/v1.0.0` contains a plugin `name:foo target:global` and a plugin `name:foo target:kubernetes`.

In this case what will happen when a user does `tanzu plugin install foo --group vmware-tkg/v1.0.0`?

cc @marckhouzam @chandrareddyp